Revision of the content structure (#727)
italia/eidas-it-wallet-docs: docs/en/architecture-overview.rst

.. include:: ../common/common_definitions.rst

Architecture Overview
=====================

The IT-Wallet System is a federated ecosystem that enables secure Digital Identity management and Digital Credential exchange for citizens and organizations. 

> The IT-Wallet ecosystem operates through a multi-layered architecture where governance bodies establish and maintain the foundational trust infrastructure, Primary Actors implement and operate the technical solutions, and external systems provide essential supporting services.

Reviewer: ```suggestion
The IT-Wallet ecosystem is built on a multi-layered architecture, where governance bodies establish and maintain the trust infrastructure, Primary Actors implement and operate the technical solutions, and external systems provide additional services.
```

> The governance level ensures that all participants in the ecosystem are compliant with security and technical standards and requirements. 

Reviewer: ```suggestion
The governance level requires all participants in the ecosystem to comply with security and technical standards and requirements.
```

> The primary operational layer implements the core Digital Credential lifecycle through coordinated interactions among specialized technical solutions. This layer operates within the trust framework established by governance actors, ensuring that all credential operations maintain security and privacy standards while enabling seamless user experiences.

Reviewer: ```suggestion
The primary operational layer implements the core Digital Credential lifecycle through coordinated interactions among specialized technical solutions, so that all credential operations maintain security and privacy standards while enabling seamless user experiences.
```

> External systems provide essential services that connect the IT-Wallet ecosystem to the national digital infrastructure. This ensures that the ecosystem is interoperable with existing government services and connected to authoritative identity and data sources.

Reviewer: ```suggestion
External systems provide services that connect the IT-Wallet ecosystem to the national digital infrastructure, enabling interoperability with existing government services and data sources.
```

> The architecture enables four fundamental interaction processes:

Reviewer: ```suggestion
The architecture enables the following interaction processes:
```

>   1. **Entity Onboarding and Federation**: This process ensures that only qualified and compliant Entities are registered within the IT-Wallet Federation. As a result of the registration process the Trust Registry is updated enabling all participants monitoring their ongoing federation status. This process may involve technical and security assessment and compliance checks to the requirements of the IT-Wallet ecosystem.
> 
>   2. **Credential Issuance**: Credential Issuers establish secure connections with Authentic Sources to request verified user attributes through standardized APIs exposed via the National Digital Data Platform. This interaction ensures that Digital Credentials are based on authoritative, up-to-date information from official registries and databases while maintaining proper authorization and audit trails.
> 
>   3. **Credential Storage and Management**: IT-Wallet Solutions receive issued Digital Credentials from authorized providers and manage them securely on user devices. This relationship enables users to obtain verified Digital Credentials from multiple sources while maintaining full control over them.
> 
>   4. **Credential Presentation and Verification**: When users need to prove their identity or specific attributes, IT-Wallet Solutions present appropriate Digital Credentials to Relying Party verification systems. This interaction enables them to validate user claims through cryptographic verification and status checks, supporting both public and private sector use cases.
> 
> The Trust Infrastructure manages the onboarding/revocation status of new Entities (with their technical solutions) and provides available Digital Credentials and schemas. It enables all ecosystem participants to discover authorized Entities and verify their current federation status.
> 
> The trust infrastructure provides automatic trust chain validation, distributed trust anchoring, standardized metadata exchange protocols, and Federation API services that enable seamless federation operations while maintaining cryptographic verification of all participant relationships.
> This approach ensures that users can trust the credentials they receive, Relying Parties can trust the credentials they verify, and all participants can trust that their counterparts have been properly vetted and remain in good standing within the federation.

Reviewer: ```suggestion
  1. **Entity Onboarding and Federation**: Only qualified and compliant entities are registered in the IT-Wallet Federation. The Trust Registry is updated during registration, allowing participants to monitor federation status. This process may include technical and security assessments.

  2. **Credential Issuance**: Credential Issuers connect to Authentic Sources via standardized APIs on the National Digital Data Platform to request verified user attributes. Digital Credentials are based on authoritative, current data with proper authorization and audit trails.

  3. **Credential Storage and Management**: IT-Wallet Solutions receive and manage Digital Credentials on user devices, allowing users to control and use credentials from multiple sources.

  4. **Credential Presentation and Verification**: Users present Digital Credentials to Relying Parties for verification. Verification systems check claims through cryptographic methods and status checks for both public and private sector use.

The Trust Infrastructure manages onboarding and revocation of entities, provides credential schemas, and lets participants discover and verify authorized entities and their status. It supports automatic trust chain validation, distributed trust anchoring, standardized metadata exchange, and Federation API services for secure, seamless federation operations.
```
